src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -252,9 +252,14 @@ internal MetricPointStatus MetricPointStatus
             this.ThrowNotSupportedMetricTypeException(nameof(this.GetHistogramSum));
         }
 
+        Debug.Assert(
+            this.mpComponents?.HistogramBuckets != null
```

Comment:
Is this check needed? The return statement that follows this checks it anyway.

Reply:
In the return we do a couple dereference ops without checks:

* `this.mpComponents!`
* `this.mpComponents.Base2ExponentialBucketHistogram!`

This assert is to make sure we have a valid path through what the return might do.

We could rewrite it like this:

```csharp
        Debug.Assert(this.mpComponents != null, "this.mpComponents was null");

        if (this.mpComponents!.HistogramBuckets != null)
        {
            return this.mpComponents!.HistogramBuckets.SnapshotSum;
        }

        Debug.Assert(this.mpComponents.Base2ExponentialBucketHistogram != null, "this.mpComponents.Base2ExponentialBucketHistogram was null");

        return this.mpComponents.Base2ExponentialBucketHistogram!.SnapshotSum;
```